core/src/validator.rs

```diff
@@ -1951,85 +1956,104 @@ fn maybe_warp_slot(
     Ok(())
 }
 
+/// Searches the blockstore for data shreds with the incorrect shred version.
 fn blockstore_contains_bad_shred_version(
     blockstore: &Blockstore,
     start_slot: Slot,
-    shred_version: u16,
-) -> bool {
-    let now = Instant::now();
+    expected_shred_version: u16,
+) -> Result<bool, BlockstoreError> {
+    const TIMEOUT_MS: u128 = 60000;
```

Comment:
```suggestion
    const TIMEOUT_MS: Duration = Duration::from_secs(60);
```

then we can just compare directly rather than doing the conversion to millis below

suggestion applies to other instances of this logic as well